src/core/agent/services/connectionService.ts

```diff
@@ -171,6 +171,16 @@ class ConnectionService extends AgentService {
       connectionMetadata.pending = false;
       connectionMetadata.status = ConnectionStatus.CONFIRMED;
     }
+    this.props.eventEmitter.emit<ConnectionStateChangedEvent>({
```

Comment:
Hmm, this was from the other PR. Same question of why it's not in the else statement. I haven't looked at this code in a while, so please explain to me what your thinking is. Maybe what you have it right.